ArjunaJTS/integration/src/main/java/com/arjuna/ats/internal/jbossatx/jta/jca/XATerminator.java

```diff
@@ -243,16 +243,6 @@ public void cancelWork (Work work, Xid xid)
 
 	@Override
 	public Transaction getTransaction(Xid xid) {
-        final XidImple xidImple;
-
-        if (xid != null && xid.getFormatId() == XATxConverter.FORMAT_ID) {
-            XidImple toImport = new XidImple(xid);
-            XATxConverter.setSubordinateNodeName(toImport.getXID(), TxControl.getXANodeName());
-            xidImple = new XidImple(toImport);
-        } else {
-            xidImple = new XidImple(xid);
-        }
-
-        return TransactionImple.getTransaction((xidImple).getTransactionUid());
+        return TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
```

Comment:
You have to check for the root transaction first as we don't want a subordinate _and_ a root transaction on one server but I think you need to check it this way (I must have missed this somewhere (maybe I included that check in the test but it would simplify the transport if we checked on its behalf you are right):
```
txn = TransactionImple.getTransaction(new XidImple(xid).getTransactionUid();
if (txn == null)
  txn = SubordinationManager.getTransactionImporter().getImportedTransaction(xid);
```